src/Control/Carrier/Parse/Measured.hs

```diff
@@ -0,0 +1,160 @@
+{-# LANGUAGE GADTs, GeneralizedNewtypeDeriving, TypeOperators, UndecidableInstances #-}
+-- | A carrier for 'Parse' effects suitable for use in production.
+module Control.Carrier.Parse.Measured
+( -- * Parse effect
+  module Control.Effect.Parse
+  -- * Parse carrier
+, ParseC(..)
+  -- * Exceptions
+, ParserCancelled(..)
+) where
+
+import qualified Assigning.Assignment as Assignment
+import qualified Assigning.Assignment.Deterministic as Deterministic
+import           Control.Effect.Error
+import           Control.Effect.Carrier
+import           Control.Effect.Parse
+import           Control.Effect.Reader
+import           Control.Effect.Trace
+import           Control.Exception
+import           Control.Monad.IO.Class
+import           Data.Blob
+import qualified Data.Error as Error
+import qualified Data.Flag as Flag
+import qualified Data.Syntax as Syntax
+import           Data.Sum
+import           Data.Term
+import           Data.Typeable
+import           Parsing.CMark
+import           Parsing.Parser
+import           Parsing.TreeSitter
+import           Prologue hiding (project)
+import           Semantic.Config
+import           Semantic.Task (TaskSession(..))
+import           Semantic.Telemetry
+import           Semantic.Timeout
+import           Source.Source (Source)
+
+newtype ParseC m a = ParseC { runParse :: m a }
+  deriving (Applicative, Functor, Monad, MonadIO)
+
+instance ( Carrier sig m
+         , Member (Error SomeException) sig
+         , Member (Reader TaskSession) sig
+         , Member Telemetry sig
+         , Member Timeout sig
+         , Member Trace sig
+         , MonadIO m
+         )
+      => Carrier (Parse :+: sig) (ParseC m) where
+  eff (L (Parse parser blob k)) = runParser blob parser >>= k
+  eff (R other) = ParseC (eff (handleCoercible other))
+
+-- | Parse a 'Blob' in 'IO'.
+runParser :: (Member (Error SomeException) sig, Member (Reader TaskSession) sig, Member Telemetry sig, Member Timeout sig, Member Trace sig, Carrier sig m, MonadIO m)
```

Comment:
In particular, the only changes I made to `runParser`/`runAssignment`/etc were to remove a couple of redundant `Member (Lift IO) sig` constraints.